modules/dnn/src/layer.cpp

```diff
@@ -84,6 +84,12 @@ Ptr<BackendNode> Layer::initTimVX(void* timVxInfo,
     return Ptr<BackendNode>();
 }
 
+Ptr<BackendNode> Layer::initCann(const std::vector<Ptr<BackendWrapper> > &inputsWrapper, const int index, const std::vector<Ptr<BackendNode> >& nodes)
+{
+    CV_Error(Error::StsNotImplemented, "CANN pipeline of " + type + " layers is not defined.");
+    return Ptr<BackendNode>();
```

Comment:
> `return Ptr<BackendNode>();`

unreachable code

Reply:
This will be useful if we have the graph split. We still need to keep it though for inheritance, don't we?

Reply:
This will be useful if we have the graph split. We still need to keep it though for inheritance, don't we?

Reply:
`CV_Error()` throws exception unconditionally.
Code after that line is dead.

There is no question about inheritance.

Reply:
I just followed what other backends do. I can do the change though in later tonight.

Reply:
OK. This should be fixed separately then for all cases.